tests/Php73/Php73Test.php

```diff
@@ -37,4 +37,30 @@ public function testIsCountableForGenerator()
 
         $this->assertFalse(is_countable(genOneToTen()));
     }
+
+    public function testHardwareTimeAsNum()
+    {
+        $hrtime = hrtime(true);
+        $this->assertTrue(is_float($hrtime) || is_int($hrtime));
+
+        usleep(1000);
+        $d1 = hrtime(true) - $hrtime;
+
+        $this->assertEquals(10000000, $d1, '', 9e6);
+    }
+
+    public function testHardwareTimeAsArray()
+    {
+        $hrtime = hrtime();
+        $this->assertInternalType('array', $hrtime);
+        $this->assertCount(2, $hrtime);
+        $this->assertInternalType('int', $hrtime[0]);
+        $this->assertInternalType('int', $hrtime[1]);
+
+        usleep(1000);
+        $hrtime2 = hrtime();
+
+        $this->assertSame(0, $hrtime2[0] - $hrtime[0]);
```

Comment:
isn't this subject to race conditions ?

Reply:
Sort of, the `usleep` call make it sleep for 0.001 of a second, if the system causes it to sleep 1000 times longer, then this might fail